ip-utils.c

```diff
@@ -12,13 +12,45 @@
  */
 
 #include <errno.h>
+#include <sys/socket.h>
 #include <sys/types.h>
 #include <arpa/inet.h>
+#include <netdb.h>
 #include <net/if.h>
 #include <string.h>
+#include <unistd.h>
+
+#include <glib.h>
 
 #include "ip-utils.h"
 
+int janus_network_detect_local_ip(char *buf, size_t buflen) {
+	struct sockaddr_in addr;
+	socklen_t len;
+	int fd = socket(AF_INET, SOCK_DGRAM, 0);
+	if(fd == -1)
+		goto error;
+	addr.sin_family = AF_INET;
+	addr.sin_port = htons(1);
+	inet_pton(AF_INET, "1.2.3.4", &addr.sin_addr.s_addr);
+	if(connect(fd, (const struct sockaddr*) &addr, sizeof(addr)) < 0)
```

Comment:
> The only purpose of this method is not just to find a local IP address, but to find an IP address that is likely to be used for communicating with the outside. 

Yes I thought so too; but unless I am much mistaken the actual IP address you are going to find will be one of the interface pointing to the default gateway. This is the correct answer if and only if the client communicates through Janus via the same network interface. It looks like in multihomed setups you can get the 'wrong' answer, for example.